File: Edge of the Ocean/almostIncreasingSequence.go
func almostIncreasingSequence(sequence []int) bool {
	screwups, tail := 0, 0
	for i := 1; i < len(sequence); i++ {
		if sequence[tail] < sequence[i] {
			tail = i
		} else {
			if tail == 0 || sequence[i] > sequence[tail - 1] {
				tail = i
			}
			screwups++
		}
	}
	return screwups == 1
}

File: CodeSignal/Edge of the Ocean/py - 5:5/almostIncreasingSequence.py
def almostIncreasingSequence(sequence):
    dropped = False
    last = prev = min(sequence) - 1
    for i in sequence:
        if i <= last:
            if dropped:
                return False
            dropped = True
            if i <= prev:
                prev = last
            else:
                prev = last = i
        else:
            prev, last = last, i
    return True